### ISSUE TITLE
Compilation failed with some assembler messages

### ISSUE THREAD

**Issue by librehat** [randombit/botan#1390]
I was trying to build botan-2.3.0 with my custom built GCC-5.5 on Debian Wheezy, and I got these errors after I configured with `../configure.py --optimize-for-size --prefix=/opt/botan2 --cc=gcc --cc-min-version=5.5` and executed `make -j2`:

```
g++ -fstack-protector -m64 -pthread -msse2 -fPIC -fvisibility=hidden -std=c++11 -D_REENTRANT -Os -momit-leaf-frame-pointer -Wall -Wextra -Wpedantic -Wstrict-aliasing -Wstrict-overflow=5 -Wcast-align -Wmissing-declarations -Wpointer-arith -Wcast-qual -Wzero-as-null-pointer-constant -Wnon-virtual-dtor -Ibuild/include -Ibuild/include/external -c ../src/lib/block/shacal2/shacal2_simd/shacal2_simd.cpp -o build/obj/lib/block_shacal2_simd.o
g++ -fstack-protector -m64 -pthread -msha -mssse3 -fPIC -fvisibility=hidden -std=c++11 -D_REENTRANT -Os -momit-leaf-frame-pointer -Wall -Wextra -Wpedantic -Wstrict-aliasing -Wstrict-overflow=5 -Wcast-align -Wmissing-declarations -Wpointer-arith -Wcast-qual -Wzero-as-null-pointer-constant -Wnon-virtual-dtor -Ibuild/include -Ibuild/include/external -c ../src/lib/block/shacal2/shacal2_x86/shacal2_x86.cpp -o build/obj/lib/block_shacal2_x86.o
g++ -fstack-protector -m64 -pthread -fPIC -fvisibility=hidden -std=c++11 -D_REENTRANT -Os -momit-leaf-frame-pointer -Wall -Wextra -Wpedantic -Wstrict-aliasing -Wstrict-overflow=5 -Wcast-align -Wmissing-declarations -Wpointer-arith -Wcast-qual -Wzero-as-null-pointer-constant -Wnon-virtual-dtor -Ibuild/include -Ibuild/include/external -c ../src/lib/block/sm4/sm4.cpp -o build/obj/lib/block_sm4.o
/tmp/ccxHIQyQ.s: Assembler messages:
/tmp/ccxHIQyQ.s:41: Error: no such instruction: `sha256rnds2 %xmm0,%xmm1,%xmm3'
/tmp/ccxHIQyQ.s:43: Error: no such instruction: `sha256rnds2 %xmm0,%xmm2,%xmm4'
/tmp/ccxHIQyQ.s:48: Error: no such instruction: `sha256rnds2 %xmm0,%xmm3,%xmm1'
/tmp/ccxHIQyQ.s:49: Error: no such instruction: `sha256rnds2 %xmm0,%xmm4,%xmm2'
/tmp/ccxHIQyQ.s:53: Error: no such instruction: `sha256rnds2 %xmm0,%xmm2,%xmm4'
/tmp/ccxHIQyQ.s:54: Error: no such instruction: `sha256rnds2 %xmm0,%xmm1,%xmm7'
/tmp/ccxHIQyQ.s:56: Error: no such instruction: `sha256rnds2 %xmm0,%xmm4,%xmm2'
/tmp/ccxHIQyQ.s:58: Error: no such instruction: `sha256rnds2 %xmm0,%xmm7,%xmm1'
/tmp/ccxHIQyQ.s:101: Error: no such instruction: `sha256rnds2 %xmm0,%xmm1,%xmm2'
/tmp/ccxHIQyQ.s:107: Error: no such instruction: `sha256rnds2 %xmm0,%xmm2,%xmm1'
/tmp/ccxHIQyQ.s:110: Error: no such instruction: `sha256rnds2 %xmm0,%xmm1,%xmm2'
/tmp/ccxHIQyQ.s:112: Error: no such instruction: `sha256rnds2 %xmm0,%xmm2,%xmm1'
make: *** [build/obj/lib/block_shacal2_x86.o] Error 1
```

I'm running it in my VM in VirtualBox. The content of `/proc/cpuinfo` is:

```
librehat@debian-vm:~/Botan-2.3.0/build$ cat /proc/cpuinfo 
processor       : 0
vendor_id       : GenuineIntel
cpu family      : 6
model           : 78
model name      : Intel(R) Core(TM) i7-6500U CPU @ 2.50GHz
stepping        : 3
cpu MHz         : 2592.012
cache size      : 4096 KB
physical id     : 0
siblings        : 2
core id         : 0
cpu cores       : 2
apicid          : 0
initial apicid  : 0
fpu             : yes
fpu_exception   : yes
cpuid level     : 22
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc pni pclmulqdq ssse3 cx16 sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx rdrand hypervisor lahf_lm abm 3dnowprefetch
bogomips        : 5184.02
clflush size    : 64
cache_alignment : 64
address sizes   : 39 bits physical, 48 bits virtual
power management:

processor       : 1
vendor_id       : GenuineIntel
cpu family      : 6
model           : 78
model name      : Intel(R) Core(TM) i7-6500U CPU @ 2.50GHz
stepping        : 3
cpu MHz         : 2592.012
cache size      : 4096 KB
physical id     : 0
siblings        : 2
core id         : 1
cpu cores       : 2
apicid          : 1
initial apicid  : 1
fpu             : yes
fpu_exception   : yes
cpuid level     : 22
wp              : yes
flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc pni pclmulqdq ssse3 cx16 sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx rdrand hypervisor lahf_lm abm 3dnowprefetch
bogomips        : 5184.02
clflush size    : 64
cache_alignment : 64
address sizes   : 39 bits physical, 48 bits virtual
power management:
```

**Comment by randombit** [comment]
It looks like your binutils is too old, so even though GCC knows how to handle the SHA-NI intrinsics, binutils doesn't know about the instructions.

You'll have to disable use of SHA-NI with `--disable-modules=shacal2_x86,sha1_x86,sha2_32_x86` (in 2.4.0 this can be done with `--disable-sha` but that wasn't available in 2.3.0)

**Comment by librehat** [comment]
Thanks that solved the issue.
